mlxtend/plotting/decision_regions.py

```diff
@@ -270,13 +281,59 @@ def plot_decision_regions(X, y, clf,
             y_data = X_highlight[feature_range_mask, y_index]
             x_data = X_highlight[feature_range_mask, x_index]
 
+        # Make sure scatter_highlight_kwargs backwards compatible defaults
+        scatter_highlight_defaults = {'c': '',
+                                      'edgecolor': 'black',
+                                      'alpha': 1.0,
+                                      'linewidths': 1,
+                                      'marker': 'o',
+                                      's': 80}
+        scatter_highlight_kwargs = format_plotting_kwargs(
+                                    default_kwargs=scatter_highlight_defaults,
+                                    user_kwargs=scatter_highlight_kwargs)
         ax.scatter(x_data,
                    y_data,
-                   c='',
-                   edgecolor='black',
-                   alpha=1.0,
-                   linewidths=1,
-                   marker='o',
-                   s=80)
+                   **scatter_highlight_kwargs)
+
+    if legend:
+        if dim > 2 and filler_feature_ranges is None:
+            pass
+        else:
+            handles, labels = ax.get_legend_handles_labels()
+            ax.legend(handles, labels,
+                      framealpha=0.3, scatterpoints=1, loc=legend)
 
     return ax
+
+
+def format_plotting_kwargs(default_kwargs=None, user_kwargs=None,
```

Comment:
That's a nice function dealing with the issues of protected keys! One little thing that comes to mind is that the way I wrote the API parser for the documentation, this would show up in the API docs. You could make this function private by prefixing it with an underscore (`_format_plotting_kwargs`), but I think this function can come in very handy in other functions (esp. plotting functions) as well in future. So, it would be nice if you could move it to the `mlxtend.utils` :)

Reply:
I don't have a better name at hand, but under a more general name like "format_kwarg_dictionaries" or so?

Reply:
Sounds good to me! I have a couple of quick questions:

1) Should this function go into its own module in `utils`, maybe `plotting.py`, or into one of the existing modules like `checking.py`?

2) Should I also add the leading underscore to make this function private in `utils` (i.e. `_format_kwarg_dictionaries` vs. `format_kwarg_dictionaries`)? I'd imagine this function will only be used for internal stuff and not directly by users, so the leading underscore would make sense. 

Thanks!

Reply:
Good points! I think we could put it into checking, because it's somewhat related to checking kwargs. I wouldn't suggest making it private, because it's generally useful (how I think about private things is that these are more like helper functions that are not useful outside the scope of a particular function and more like a side effect of refactoring, whereas the public ones are more broadly useful).